packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -0,0 +1,539 @@
+        error: productsError
+    } = useProducts(
+        {parameters: {ids: productIds, allImages: true}},
+        {
+            enabled: Boolean(productIds),
+            select: (data) => {
+                return (
+                    data?.data?.reduce((acc, p) => {
+                        acc[p.id] = p
+                        return acc
+                    }, {}) || {}
+                )
+            }
+        }
+    )
+    const {data: customer} = useCurrentCustomer()
+    const addresses = customer?.addresses || []
+    const [selectedAddresses, setSelectedAddresses] = useState({})
+
+    if (!basket?.productItems?.length) {
+        return (
+            <Center
+                p={8}
+                textAlign="center"
+                color="gray.500"
+                role="status"
+                aria-live="polite"
+                aria-label={formatMessage(noItemsInBasketMessage)}
+            >
+                <VStack spacing={4}>
+                    <Text fontSize="lg" fontWeight="medium">
+                        {formatMessage(noItemsInBasketMessage)}
+                    </Text>
+                </VStack>
+            </Center>
+        )
+    }
+
+    // Loading state
+    if (productsLoading) {
+        return (
+            <Center p={8} textAlign="center" color="gray.500">
+                <VStack spacing={4}>
+                    <Spinner size="lg" />
+                    <Text>
+                        {formatMessage({
+                            id: 'shipping_multi_address.loading.message',
+                            defaultMessage: 'Loading products...'
+                        })}
+                    </Text>
+                </VStack>
+            </Center>
+        )
+    }
+
+    // Error state
+    if (productsError) {
+        return (
+            <Alert
+                status="error"
+                variant="subtle"
+                flexDirection="column"
+                alignItems="center"
+                justifyContent="center"
+                textAlign="center"
+                height="200px"
+                aria-live="assertive"
+            >
+                <AlertIcon boxSize="40px" mr={0} />
+                <AlertTitle mr={2}>
+                    {formatMessage({
+                        id: 'shipping_multi_address.error.label',
+                        defaultMessage: 'Error loading products'
+                    })}
+                </AlertTitle>
+                <AlertDescription>
+                    {formatMessage({
+                        id: 'shipping_multi_address.error.message',
+                        defaultMessage: 'Error loading products. Please try again.'
+                    })}
+                </AlertDescription>
+            </Alert>
+        )
+    }
+
+    const onAddNewAddress = () => {
+        // TODO: Implement add new address logic/modal
+        console.log('Add New Address clicked!')
+    }
+
+    return (
+        <Box
+            role="region"
```

Comment:
Nested "region"

The issue: This creates confusing navigation landmarks where screen readers will announce multiple nested regions, making it harder for users to understand the page structure. Better approach: Remove the parent role="region" and keep only the two meaningful child regions, or use the parent region and remove the child ones - don't nest them.